erpnext/controllers/transaction_controller.py

```diff
@@ -736,6 +736,12 @@ def set_taxes_and_charges(self):
 		tax_master_doctype = self.meta.get_field("taxes_and_charges").options
 
 		if (self.is_new() or self.is_pos_profile_changed()) and not self.get("taxes"):
+			if self.company and self.get("customer"):
+				# get default tax template from customer master
+				customer_default_taxes_and_charges = frappe.get_cached_value("Customer", self.get("customer"), "taxes_and_charges_template")
+				if customer_default_taxes_and_charges:
+					self.taxes_and_charges = customer_default_taxes_and_charges
+			
```

Comment:
still duplicated logic .. see if you can use `accounts.party.set_taxes` method instead of rewriting the same logic